feat: add new authorizer cmds

iljaSL opened the pull request.

iljassh committed:
feat: add new authorizer cmds

cuongssh commented on cmd/principals.go:
```diff
@@ -0,0 +1,266 @@
+
+	cmd := &cobra.Command{
+		Use:   "sign",
+		Short: "Get a signature",
```
Sign data using principal private key?

cuongssh reviewed (changes requested):
The files are OK except webproxy lacks the configuration file donwload. 

It got a bit complicated as extender, web-proxy and carrier are trusted clients that also have data fetched through local-user-store micro-service. The List, Show sub-commands are actually list the CAs instead of showing  or listing the component's data.

I suggest, in addition to the missing conf download, the following:
- Save list and show sub-command for the calls to local-user-store to pull out the component data. Change the current list and show sub-command to list-ca and show-ca. 
- Similarly, revocation-list can be shortened to show-crl 
- webproxy consists of a web-proxy component and a carrier component. If the whole package is addressed by a single command "web-proxy"  then make sure it returns data from both components, or from a specific component if asked. If the whole package is addressed separately by "web-proxy" and "carrier" commands then add carrier.go

iljassh committed:
feat: add pre config cmds

iljassh committed:
refactor: restruct trusted client cmds into one file

iljaSL commented:
Restructured trusted client commands into one file -> extender.go
User can now choose with the `type` flag for a specific component.

cuongssh commented on cmd/extender.go:
```diff
@@ -0,0 +1,416 @@
+//
+func extenderCmd() *cobra.Command {
+	cmd := &cobra.Command{
+		Use:          "extender",
```
Based on the usage it seems that the top-most command should be "clients" or "trusted-clients" as it covers extender, web-proxy and carrier now.

iljassh committed:
rf: rename cmd and file

iljassh committed:
rf: rename func, vars, structs to trusted client

iljaSL commented:
Renamed the `extender` command to `trusted-clients`, as `clients` was already in use.
Renamed also the file, functions, variables etc. according to the command name change.
